src/main/java/com/example/demo/services/ILocationService.java

```diff
@@ -0,0 +1,16 @@
+package com.example.demo.services;
+
+import com.example.demo.entities.LocationEntity;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+
+import java.util.Optional;
+
+public interface ILocationService {
```

Comment:
Interface is fine, but in Spring they are optional if you only have the one implementation, which can make life a little simpler